Resolve Pitest Issues - AvoidStarImportCheck (1)

Child issue of https://github.com/checkstyle/checkstyle/issues/7797 ,

This issue specifically focuses on the line https://github.com/checkstyle/checkstyle/blob/abf829f542023e089c741c55f73098cae52dcb12/.ci/pitest.sh#L74

@rnveach I got proof!

I  just ran `java -jar checkstyle-8.29-all.jar -c config.xml Test.java` on the following source file:
```java
import java.util.*; // a class import with a star

class Foo{
  private int a;
  /* public comment */ // OK, comment is ignored
  public bar1() {} // violation
  public bar2() {} // violation
}
``
For each of the two following configs:
```xml
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
		<module name="AvoidStarImport"/>
```
And 
```xml
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
		<module name="AvoidStarImport">
  <property name="allowClassImports" value="true"/>
</module>
```

the results are respectively:

```
Starting audit...
Audit done.
```
And 

```
Starting audit...
[ERROR] /var/tmp/Test.java:2: Using the '.*' form of import should be avoided - java.util.*. [AvoidStarImport]
Audit done.
Checkstyle ends with 1 errors.
```

I believe this prove my point.

--- a/.ci/pitest.sh
+++ b/.ci/pitest.sh
@@ -71,7 +71,6 @@ pitest-header)
 pitest-imports)
   mvn -e -P$1 clean test org.pitest:pitest-maven:mutationCoverage;
   declare -a ignoredItems=(
-  "AvoidStarImportCheck.java.html:<td class='covered'><pre><span  class='survived'>            &#38;&#38; ast.getType() == TokenTypes.STATIC_IMPORT) {</span></pre></td></tr>"
   "CustomImportOrderCheck.java.html:<td class='covered'><pre><span  class='survived'>        else if (customImportOrderRules.contains(SAME_PACKAGE_RULE_GROUP)) {</span></pre></td></tr>"
   "CustomImportOrderCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (bestMatch.group.equals(NON_GROUP_RULE_GROUP)) {</span></pre></td></tr>"
   "CustomImportOrderCheck.java.html:<td class='covered'><pre><span  class='survived'>            if (customImportOrderRules.contains(SAME_PACKAGE_RULE_GROUP)) {</span></pre></td></tr>"

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStarImportCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStarImportCheck.java
@@ -216,12 +216,13 @@ public class AvoidStarImportCheck
 
     @Override
     public void visitToken(final DetailAST ast) {
-        if (!allowClassImports && ast.getType() == TokenTypes.IMPORT) {
-            final DetailAST startingDot = ast.getFirstChild();
-            logsStarredImportViolation(startingDot);
+        if (ast.getType() == TokenTypes.IMPORT) {
+            if (!allowClassImports) {
+                final DetailAST startingDot = ast.getFirstChild();
+                logsStarredImportViolation(startingDot);
+            }
         }
-        else if (!allowStaticMemberImports
-            && ast.getType() == TokenTypes.STATIC_IMPORT) {
+        else if (!allowStaticMemberImports) {
             // must navigate past the static keyword
             final DetailAST startingDot = ast.getFirstChild().getNextSibling();
             logsStarredImportViolation(startingDot);

